### ISSUE TITLE
Nonce does not match on login

### ISSUE THREAD

**Issue by caleteeter** [microsoft/vscode-azure-account#142]
When calling Azure: SignIn, I am getting this nonce match issue, which prevents successful login.  I have tried other browser, clearing cache, reinstalling VS Code, etc. with nothing resolving this.

Environment:

OS: Windows 10 Enterprise 1903 (18362.175)
VSCode: Visual Studio Code - Insiders 1.36.0-insider

On a side note, I have side loaded the retail version of VS Code (1.35) which works fine.

**Comment by chrmarti** [comment]
@jrieken Did anything change on the `openExternal` side? The request shows a double-encoded path with Insiders, e.g.: `"/signin?nonce=BX8WqKptWGArQ9zYCWokhg%253D%253D"`. Decoding that gives `"BX8WqKptWGArQ9zYCWokhg%3D%3D"` when `"BX8WqKptWGArQ9zYCWokhg=="` is expected.

**Comment by jrieken** [comment]
Yeah, the URI implementation changed so that it doesn't encode query strings of http/https/ftp uris anymore. Tho that should mean no encoding, not double encoding of certain characters. What's the full uri?

**Comment by chrmarti** [comment]
I do `Uri.parse('http://localhost:60371/signin?nonce=iiK1zRI%2BHyDCKb2zatvrYA%3D%3D')`.

Presumably this now assumes the query to not be encoded. 🐛

@jrieken Is this something you are going to look into or should I (and maybe other extensions) catch up with the change?
